code/go/0chain.net/blobbercore/allocation/workers.go

```diff
@@ -239,6 +239,19 @@ func updateAllocationInDB(ctx context.Context, a *Allocation, sa *transaction.St
 		}
 	}
 
+	_, err = lru.Get(a.ID)
+	if err == nil {
+		errInRemovingAllocationFromLRU := lru.Delete(a.ID)
+		if errInRemovingAllocationFromLRU != nil {
+			return nil, err
+		}
+	}
+
+	err = lru.Add(a.ID, a)
```

Comment:
most likely we should not remove it, we can simply add and it should be replaced

Reply:
Let me try it out